libraries/AP_Frsky_Telem/frsky.cpp

```diff
@@ -0,0 +1,232 @@
+            if (packet[i] != previous_packet[i]) {
+                equal = false;
+                // update previous_packet
+                memcpy(previous_packet,packet,SPORT_PACKET_SIZE);
+                break;
+            }
+        }
+        if (equal) {
+            return false;
+        }
+        */
+    }
+#endif
+    //check CRC
+    int16_t crc = 0;
+    for (uint8_t i=1; i<SPORT_PACKET_SIZE; ++i) {
+        crc += packet[i]; // 0-1FE
+        crc += crc >> 8;  // 0-1FF
+        crc &= 0x00ff;    // 0-FF
+    }
+    return (crc == 0x00ff);
+}
+
+/*
+ * Calculates the sensor id from the physical sensor index [0-27]
+        0x00, 	// Physical ID 0 - Vario2 (altimeter high precision)
+        0xA1, 	// Physical ID 1 - FLVSS Lipo sensor
+        0x22, 	// Physical ID 2 - FAS-40S current sensor
+        0x83, 	// Physical ID 3 - GPS / altimeter (normal precision)
+        0xE4, 	// Physical ID 4 - RPM
+        0x45, 	// Physical ID 5 - SP2UART(Host)
+        0xC6, 	// Physical ID 6 - SPUART(Remote)
+        0x67, 	// Physical ID 7 - Ardupilot/Betaflight EXTRA DOWNLINK
+        0x48, 	// Physical ID 8 -
+        0xE9, 	// Physical ID 9 -
+        0x6A, 	// Physical ID 10 -
+        0xCB, 	// Physical ID 11 -
+        0xAC, 	// Physical ID 12 -
+        0x0D, 	// Physical ID 13 - Ardupilot/Betaflight UPLINK
+        0x8E, 	// Physical ID 14 -
+        0x2F, 	// Physical ID 15 -
+        0xD0, 	// Physical ID 16 -
+        0x71, 	// Physical ID 17 -
+        0xF2, 	// Physical ID 18 -
+        0x53, 	// Physical ID 19 -
+        0x34, 	// Physical ID 20 - Ardupilot/Betaflight EXTRA DOWNLINK
+        0x95, 	// Physical ID 21 -
+        0x16, 	// Physical ID 22 - GAS Suite
+        0xB7, 	// Physical ID 23 - IMU ACC (x,y,z)
+        0x98, 	// Physical ID 24 -
+        0x39, 	// Physical ID 25 - Power Box
+        0xBA 	// Physical ID 26 - Temp
+        0x1B	// Physical ID 27 - ArduPilot/Betaflight DEFAULT DOWNLINK
+ * for FrSky SPort Passthrough (OpenTX) protocol (X-receivers)
+ */
+#define BIT(x, index) (((x) >> index) & 0x01)
+uint8_t FRSky::get_sport_sensor_id(uint8_t physical_id)
+{
+    uint8_t result = physical_id;
+    result += (BIT(physical_id, 0) ^ BIT(physical_id, 1) ^ BIT(physical_id, 2)) << 5;
+    result += (BIT(physical_id, 2) ^ BIT(physical_id, 3) ^ BIT(physical_id, 4)) << 6;
+    result += (BIT(physical_id, 0) ^ BIT(physical_id, 2) ^ BIT(physical_id, 4)) << 7;
+    return result;
+}
+
+/*
+  send 1 byte and do byte stuffing
+*/
+void FRSky::send_byte(AP_HAL::UARTDriver *_port, uint8_t byte)
+{
+    if (byte == START_STOP_D) {
+        _port->write(0x5D);
+        _port->write(0x3E);
+    } else if (byte == BYTESTUFF_D) {
+        _port->write(0x5D);
+        _port->write(0x3D);
+    } else {
+        _port->write(byte);
+    }
+}
+
+/*
+ * send one uint16 frame of FrSky data - for FrSky D protocol (D-receivers)
+ */
+void  FRSky::send_uint16(AP_HAL::UARTDriver *_port, uint16_t id, uint16_t data)
```

Comment:
I'm really not convinced by the need of having this new class to do this.

AFAICS this doesn't need to move from where it is.

The parsing should be a separate parse-state-holding object, 'though.  Not called "FRSky" :-)